### PR TITLE
wrong response when delete conversation by api

### LINKED ISSUE #5400: wrong response when delete conversation by api
### Self Checks

- [X] This is only for bug report, if you would like to ask a question, please head to [Discussions](https://github.com/langgenius/dify/discussions/categories/general).
- [X] I have searched for existing issues [search for existing issues](https://github.com/langgenius/dify/issues), including closed ones.
- [X] I confirm that I am using English to submit this report (我已阅读并同意 [Language Policy](https://github.com/langgenius/dify/issues/1542)).
- [X] Please do not modify this template :) and fill in all the required fields.

### Dify version

0.6.11

### Cloud or Self Hosted

Self Hosted (Docker), Self Hosted (Source)

### Steps to reproduce

when I use the api of deleting conversation, the content of response is empty, and http response code is 204.
image
image

### ✔️ Expected Behavior

I need the content of response is as the document says
image

### ❌ Actual Behavior

the content of response is empty, and http response code is 204.
image

### PR DESCRIPTION
**Pull request by chivalry1314**
# Description

wrong response when delete conversation by api

Fixes #5400

## Type of Change

- [x] Bug fix (non-breaking change which fixes an issue)

# How Has This Been Tested?

image
image
image

- [ ] TODO

# Suggested Checklist:

- [x] I have performed a self-review of my own code
- [x] I have commented my code, particularly in hard-to-understand areas
- [x] My changes generate no new warnings
- [x] I ran `dev/reformat`(backend) and `cd web && npx lint-staged`(frontend) to appease the lint gods
- [x] `optional` I have made corresponding changes to the documentation 
- [x] `optional` I have added tests that prove my fix is effective or that my feature works
- [x] `optional` New and existing unit tests pass locally with my changes

### CODE REVIEW

**Review by crazywoola — DISMISSED** [review]

**Comment by takatost** [comment]
https://developer.mozilla.org/en-US/docs/Web/HTTP/Methods/DELETE
Returning a 204 status code is reasonable and compliant with the standards. You can refer to this specification link for more details.

**Comment by chivalry1314** [comment]
> https://developer.mozilla.org/en-US/docs/Web/HTTP/Methods/DELETE Returning a 204 status code is reasonable and compliant with the standards. You can refer to this specification link for more details.

but if I use 204 status code I cannot get the success message as the dify document says from the server like the first picture I offered. the status code 204 means no further information is to be supplied.

**Comment by takatost** [comment]
> > https://developer.mozilla.org/en-US/docs/Web/HTTP/Methods/DELETE Returning a 204 status code is reasonable and compliant with the standards. You can refer to this specification link for more details.
> 
> 
> 
> but if I use 204 status code I cannot get the success message as the dify document says from the server like the first picture I offered. the status code 204 means no further information is to be supplied. 

This is correct, so the direction for the fix could be to remove the response content from the API implementation and documentation, while keeping the 204 status code. Thank you.

File: .github/workflows/build-push.yml
Comment (crazywoola) [inline .github/workflows/build-push.yml:1]: Please remain this file untouched in main branch.

File: api/controllers/service_api/app/message.py
Comment (crazywoola) [inline api/controllers/service_api/app/message.py:1]: This is not related as well.

**Review by crazywoola — CHANGES_REQUESTED** [review]

File: api/services/message_service.py [hunk 1]
```diff
@@ -69,6 +70,13 @@ def pagination_by_first_id(cls, app_model: App, user: Optional[Union[Account, En
 
         history_messages = list(reversed(history_messages))
 
+        # 进行特殊处理，截取“问询”字段
+        for hismessage in history_messages:
```
Comment (takatost) [inline api/services/message_service.py:74]: Please remove this particular piece of logic

File: api/controllers/service_api/app/message.py [hunk 2]
```diff
@@ -58,7 +58,9 @@ class MessageListApi(Resource):
         'conversation_id': fields.String,
         'inputs': fields.Raw,
         'query': fields.String,
+        'message_tokens': fields.Integer,
```
Comment (takatost) [inline api/controllers/service_api/app/message.py:61]: The changes here are not in line with the topic of current PR, please create a new PR separately, thanks!

File: .github/workflows/build-push.yml [hunk 3]
```diff
@@ -2,11 +2,7 @@ name: Build and Push API & Web
 
 on:
   push:
-    branches:
-      - "main"
-      - "deploy/dev"
-  release:
-    types: [published]
+    branches: [ main ]
```
Comment (takatost) [inline .github/workflows/build-push.yml:5]: Please restore this workflow to its original state.

**Comment by takatost** [comment]
> > https://developer.mozilla.org/en-US/docs/Web/HTTP/Methods/DELETE Returning a 204 status code is reasonable and compliant with the standards. You can refer to this specification link for more details.
> 
> but if I use 204 status code I cannot get the success message as the dify document says from the server like the first picture I offered. the status code 204 means no further information is to be supplied.

204 status code of means the execution was successful, so you can ignore the `success` message internal.

**Review by takatost — CHANGES_REQUESTED** [review]

**Comment by crazywoola** [comment]
Please resolve those comments and reopen a pull request based on the main branch.

**Pull request closed without merging**